pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -219,6 +221,10 @@ protected void doAddRecord(MutableSegment segment, RecordInfo recordInfo) {
   @Override
   protected GenericRow doUpdateRecord(GenericRow record, RecordInfo recordInfo) {
     assert _partialUpsertHandler != null;
+    // For Overwrite merger, it respects the incoming values unless it's null. We can use this to optimize disk read.
+    if (UpsertConfig.Strategy.OVERWRITE.equals(_partialUpsertHandler.getDefaultPartialUpsertMerger())) {
```

Comment:
Are you comparing a merger with an enum?

Reply:
good catch, thanks! I though the defaultMerger in upsertHandler is a enum. Let me update this part.